design/src/docs/pokemon_service.md

```diff
@@ -0,0 +1,239 @@
+
+Finally it creates a RestJSON `Router`, because that is the service's protocol.
+You will have noticed, each operation is implemented as a `pub async fn`. Each operation is wrapped into an `OperationHandler`, generated [here](https://github.com/awslabs/smithy-rs/blob/db48039065bec890ef387385773b37154b555b14/codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ServerOperationHandlerGenerator.kt#L30).
+`OperationHandler` implements tower's `Service` [trait](https://docs.rs/tower-service/latest/tower_service/trait.Service.html). Implementing `Service` means that
+the business logic is written as protocol-agnostic and clients request a service by calling into them, similar to an RPC call.
+```rust
+aws_smithy_http_server::routing::Router::new_rest_json_router(vec![
+    {
+        let svc = crate::server_operation_handler_trait::operation(
+            registry.capture_pokemon_operation,
+        );
+```
+At this level, logging might be prohibited by the `@sensitive` trait. If there are no `@sensitive` shapes, the generated code looks like:
+```rust
+let request_fmt = aws_smithy_http_server::logging::sensitivity::RequestFmt::new();
+let response_fmt = aws_smithy_http_server::logging::sensitivity::ResponseFmt::new();
+let svc = aws_smithy_http_server::logging::InstrumentOperation::new(
+    svc,
+    "capture_pokemon_operation",
+)
+.request_fmt(request_fmt)
+.response_fmt(response_fmt);
+```
+Accessing the Pokédex is modeled as a restricted operation: a passcode is needed by the Pokémon trainer.
+Not to log the passcode, the code will be generated [here](https://github.com/awslabs/smithy-rs/blob/db48039065bec890ef387385773b37154b555b14/codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ServerHttpSensitivityGenerator.kt#L58) as:
+```rust
+let request_fmt = aws_smithy_http_server::logging::sensitivity::RequestFmt::new()
+    .header(|name: &http::header::HeaderName| {
+        #[allow(unused_variables)]
+        let name = name.as_str();
+        let name_match = matches!(name, "passcode");
+        let key_suffix = None;
+        let value = name_match;
+        aws_smithy_http_server::logging::sensitivity::headers::HeaderMarker {
+            value,
+            key_suffix,
+        }
+    })
+    .label(|index: usize| matches!(index, 1));
+let response_fmt = aws_smithy_http_server::logging::sensitivity::ResponseFmt::new();
+```
+Each route is a pair, [`BoxCloneService`](https://docs.rs/tower/latest/tower/util/struct.BoxCloneService.html) wrapping the service operation (the implementation) and
+the information to consume the service operation.
+```rust
+(tower::util::BoxCloneService::new(svc), capture_pokemon_operation_request_spec)
+```
+Now the `Router` is built. `Router` is not code generated, instead lives in the aws-smithy-http-server crate; it is, whenever possible, better to write Rust code in Rust, rather than generate it to aid development.
```

Comment:
```suggestion
Now the `Router` is built. `Router` is not code generated, it instead lives in the [`aws-smithy-http-server`](https://docs.rs/aws-smithy-http-server/latest/aws_smithy_http_server/) crate; it is, whenever possible, better to write Rust code in Rust, rather than generate it to aid development.
```

Reply:
> better to write Rust code in Rust, rather than generate it to aid development.

This is true, although only one (arguably small) factor in deciding where to put code. I think the primary and motivating dichotomy that drives our decision-making regarding where to place code is: "If the code is doing something **service-specific** i.e. it only applies to the service modeled in Smithy, then that code should be code-generated. If the code is **service-agnostic** i.e. it applies to _all_ services that we generate, or to a large amount of them (i.e. all services using protocol X), it should then live in the runtime".

This is of course not a hard and fast rule. There are grey areas: for example, I feel like whether we place code in the [inlineable pseudo-crate](https://github.com/awslabs/smithy-rs/tree/main/rust-runtime/inlineable) or in codegen is more driven by maintainability concerns as you point out.